spynnaker/pyNN/models/neural_projections/connectors/abstract_connector.py

```diff
@@ -96,6 +98,7 @@ def set_projection_information(self, machine_time_step, synapse_info):
         """
         self._rng = (self._rng or NumpyRNG())
         self.__min_delay = machine_time_step / MICRO_TO_MILLISECOND_CONVERSION
+        self.__synapse_info = synapse_info
```

Comment:
Good question, but (a) I thought that we had a test for this, so it should show such problems? and (b) I'm not sure we should necessarily be advocating the sharing of connectors...

Reply:
(The tests I thought were dealing with such a scenario are in spynnaker_integration_tests/test_reuse_connector/)

Reply:
We can't stop people reusing connectors as this is part of PyNN.  There is a test, but then none of the tests use the new distance dependence weights and delays.  From what I can see, the synapse info is used to get the pre and post population positions, so the test would be to try the same connector but a different set of pre and post populations with different positions in each.

Reply:
The solution is that the synapse_info is likely available at the time of the call for these functions, so passing that in rather than storing it should work.

Reply:
I've just tried this and it seems to work as I would expect it to with distance-dependent weights and delays and a reused connector on different pre- and post-populations.  The only explanation I can come up with is that it does each set of calculations independently somehow...

Reply:
Can you add this as a test here?